cmd/eudico/mirvalidator/utils.go

```diff
@@ -138,10 +138,11 @@ func newLp2pHost(dir string) (host.Host, error) {
 			libp2p.Identity(pk),
 			libp2p.DefaultTransports,
 			libp2p.ListenAddrStrings(
-				"/ip4/0.0.0.0/tcp/0",
-				"/ip6/::/tcp/0",
-				"/ip4/0.0.0.0/udp/0/quic",
-				"/ip6/::/udp/0/quic"),
+				fmt.Sprintf("/ip4/0.0.0.0/tcp/%d", port),
+				fmt.Sprintf("/ip6/::/tcp/%d", port),
+				fmt.Sprintf("/ip4/0.0.0.0/udp/%d/quic", port),
+				fmt.Sprintf("/ip6/::/udp/%d/quic", port),
+			),
```

Comment:
Have you tested this? I don't think you can set the same in all of them or can you? I think they will collide, you need a different one for each one. I would set all to 0 and allow the configuration of the udp and tcp.